services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/ValidateFlowAction.java

```diff
@@ -66,29 +67,31 @@ public ValidateFlowAction(PersistenceManager persistenceManager, FlowOperationsD
                 new HashSet<>(Optional.ofNullable(context.getPathsToReroute()).orElse(emptySet()));
         dashboardLogger.onFlowPathReroute(flowId, pathsToReroute, context.isForceReroute());
 
+        final boolean forceDefaultEncapsulation = featureTogglesRepository.find()
+                .map(FeatureToggles::getFlowsRerouteUsingDefaultEncapType)
+                .orElseGet(FeatureToggles.DEFAULTS::getFlowsRerouteUsingDefaultEncapType);
+
         Flow flow = persistenceManager.getTransactionManager().doInTransaction(() -> {
-            Flow foundFlow = getFlow(flowId, FetchStrategy.NO_RELATIONS);
+            Flow foundFlow = getFlow(flowId, FetchStrategy.DIRECT_RELATIONS);
             if (foundFlow.getStatus() == FlowStatus.IN_PROGRESS) {
                 throw new FlowProcessingException(ErrorType.REQUEST_INVALID,
                         format("Flow %s is in progress now", flowId));
             }
 
+            stateMachine.setOriginalFlow(new Flow(foundFlow));
+
             stateMachine.setOriginalFlowStatus(foundFlow.getStatus());
-            stateMachine.setOriginalEncapsulationType(foundFlow.getEncapsulationType());
             stateMachine.setRecreateIfSamePath(!foundFlow.isActive() || context.isForceReroute());
 
-            flowRepository.updateStatus(foundFlow.getFlowId(), FlowStatus.IN_PROGRESS);
```

Comment:
`updateStatus()` also updates `time_modify` field. `createOrUpdate()` wouldn't do it